org.scala-ide.sdt.core/src/org/scalaide/core/internal/builder/zinc/EclipseSbtBuildManager.scala

```diff
@@ -55,7 +58,8 @@ class EclipseSbtBuildManager(val project: IScalaProject, settings0: Settings) ex
   private val sources: mutable.Set[IFile] = mutable.Set.empty
   private val cached = new AtomicReference[SoftReference[Analysis]]
 
-  private val cachePath = project.underlying.getFile(".cache")
+  private val uniqueCacheExtension = UUID.randomUUID.toString
```

Comment:
Why random? Why not use `macro`, `main` or `test`?

Is this working between two Eclipse sessions? It seems to me this code will never find its own cache after a restart, since the name will be each time different. The cache file should be reused, that's why it's persisted, no?